### PR TITLE
Add assert_data acceptance method

### PR DESCRIPTION
**Pull request by blowmage**
Extract and simplify data checks to an assertion method.

[refs #3084]

### CODE REVIEW

File: google-cloud-bigquery/acceptance/bigquery_helper.rb [hunk 1]
```diff
@@ -123,6 +123,14 @@ def random_file_destination_name
       "kitten-test-data-#{SecureRandom.hex}.json"
     end
 
+    def assert_data data
+      assert_equal Google::Cloud::Bigquery::Data, data.class
+      refute_nil data.kind
+      refute_nil data.etag
+      refute_nil data.total
+      refute_nil data.count
```
Comment (quartzmo) [inline google-cloud-bigquery/acceptance/bigquery_helper.rb]: Why the change now to basic assertions?

Comment (blowmage) [reply google-cloud-bigquery/acceptance/bigquery_helper.rb]: Because it is simpler and I didn't want to fight the Spec DSL to get these tests passing again.
